### ISSUE TITLE
spurious "unused extern crate"

### ISSUE THREAD

**Issue by RalfJung** [rust-lang/rust#56098]
When compiling this code
```rust
#![feature(libc, extern_crate_item_prelude)]

extern crate libc;

use std::ffi::CString;

mod mlibc {
    use libc::{c_char, size_t};
    extern {
        #[link_name = "strlen"]
        pub fn my_strlen(str: *const c_char) -> size_t;
    }
}

fn strlen(str: String) -> usize {
    // C string is terminated with a zero
    let s = CString::new(str).unwrap();
    unsafe {
        mlibc::my_strlen(s.as_ptr()) as usize
    }
}

pub fn main() {
    let len = strlen("Rust".to_string());
    assert_eq!(len, 4);
}
```
with `--edition 2018 -Wunused`, I get an incorrect warning:
```
warning: unused extern crate
  --> tests/run-pass-fullmir/foreign-fn-linkname.rs:15:1
   |
15 | extern crate libc;
   | ^^^^^^^^^^^^^^^^^^ help: remove it
   |
   = note: `-W unused-extern-crates` implied by `-W unused`
```
The `extern crate` *is* used; removing it breaks the program.

**Comment by RalfJung** [comment]
Yes, this is with
```
rustc 1.32.0-nightly (5aff30734 2018-11-19)
```

**Comment by RalfJung** [comment]
Note that I need to add `-Wunused`. No warning is shown without that flag.

**Comment by Mark-Simulacrum** [comment]
I can reproduce on playground with `#![warn(unused)]` as well: https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=f042e0d802065bb01c52be2d274f00b6

**Comment by RalfJung** [comment]
I cannot reproduce the warning any more, it seems to be fixed.

However, I am not sure if there is a testcase for that.

**Comment by kadiwa4** [comment]
@rustbot label: +E-needs-test
Test could be added to `tests/ui/foreign/foreign-fn-linkname.rs` (which is the example above).
The `#[allow(unused_extern_crates)]` in `std` can probably also be removed

**Comment by Nilstrieb** [comment]
Unassigning because they told me that they don't have the time to do this.
